front-end/src/views/EditView.vue

```diff
@@ -0,0 +1,72 @@
+<script setup>
+import ProgressSpinner from 'primevue/progressspinner';
+import UsersForm from '../components/UsersForm.vue';
+import WordsForm from '../components/WordsForm.vue';
+import { computed } from  'vue';
+import { useAPIGatewayStore } from '../stores/apiGateway';
+
+const apiGateway = useAPIGatewayStore();
+
+const currentRoute = computed(() => {
+  return apiGateway.currentRoute;
+});
+
+const title = computed(() => {
+  return `ADD ${currentRoute.value.name.toUpperCase()}`;
+});
+
+const loading = computed(() => {
+  return apiGateway.isLoading;
+});
+
+const addData = (payload) => {
+  apiGateway.postData(payload);
+}
+</script>
+
+<template>
+  <div class='edit-wrapper'>
+    <div class='title-wrapper'>
+      <h1 class='title'>{{ title }}</h1>
+    </div>
+    <div v-if='loading' class='spinner-wrapper'>
+      <ProgressSpinner style='width: 50px; height: 50px;' strokeWidth='5' fill='var(--surface-ground)'></ProgressSpinner>
```

Comment:
a point up for discussion when we meet again. Could this be moved somewhere globally and triggered by any axios traffic? Have you ever worked with interceptors in axios?

Reply:
I have not used axios interceptors yet but they look interesting. I assume you are referencing store variable "loading" used in the "v-if='loading'" part. I could look into using global interceptors to accomplish this if you would like.

Reply:
so a Spinner would be added on App.Vue

could be bound to a property in store. 

interceptor for all axios traffic will set the property in store to true/false

You don't have to do it, just a point for discussion